src/mlpack/tests/cf_test.cpp

```diff
@@ -1165,4 +1039,70 @@ BOOST_AUTO_TEST_CASE(SerializationCombinedNormalizationTest)
                     ItemMeanNormalization>>();
 }
 
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * EuclideanSearch.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictEuclideanSearch)
+{
+  CFPredict<NMFPolicy, OverallMeanNormalization, EuclideanSearch>();
+}
+
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * CosineSearch.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictCosineSearch)
+{
+  CFPredict<NMFPolicy, OverallMeanNormalization, CosineSearch>();
+}
+
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * PearsonSearch.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictPearsonSearch)
+{
+  CFPredict<NMFPolicy, OverallMeanNormalization, PearsonSearch>();
+}
+
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * AverageInterpolation.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictAverageInterpolation)
+{
+  CFPredict<NMFPolicy,
+            OverallMeanNormalization,
+            EuclideanSearch,
+            AverageInterpolation>();
+}
+
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * SimilarityInterpolation.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictSimilarityInterpolation)
+{
+  CFPredict<NMFPolicy,
+            OverallMeanNormalization,
+            EuclideanSearch,
+            SimilarityInterpolation>();
+}
+
+/**
+ * Make sure that Predict() is returning reasonable results for
+ * RegressionInterpolation.
+ */
+BOOST_AUTO_TEST_CASE(CFPredictRegressionInterpolation)
+{
+  // The small Grouplens dataset we use is very sparse.
+  // RegressionInterpolation doesn't generalize well when the data
+  // is very sparse. Thus we set a very high rmse bound here.
+  CFPredict<NMFPolicy,
+            OverallMeanNormalization,
+            EuclideanSearch,
+            RegressionInterpolation>(12);
```

Comment:
How do you think, maybe it is reasonable to use another dataset for this test?

Reply:
Yeah that would be better. But the problem is I didn't find a dataset which is small and dense enough to be suitable for the test here.

Reply:
@lozhnikov Actually I just found that changing from `NMFPolicy` to `RegSVDPolicy` can reduce `rmse` to around `1.50`. So we can just use `RegSVDPolicy` instead of `NMFPolicy` here.

Reply:
@Wenhao-H Let's wait for the test results. Actually, I just wanted to point out some interesting questions. So, we could merge the PR, open an issue and investigate the questions later.

Reply:
Sure!